src/main/plugins/control-panel-plugin/control-panel-plugin.ts

```diff
@@ -0,0 +1,63 @@
+import { SearchResultItem } from "../../../common/search-result-item";
+import { PluginType } from "../../plugin-type";
+import { IconType } from "../../../common/icon/icon-type";
+import { TranslationSet } from "../../../common/translation/translation-set";
+import { UserConfigOptions } from "../../../common/config/user-config-options";
+import { SearchPlugin } from "../../search-plugin";
+import { ControlPanelOptions } from "../../../common/config/control-panel-options";
+import { ControlPanelItem } from "./control-panel-item";
+import { defaultControlPanelIcon } from "../../../common/icon/default-icons";
+import { ControlPanelItemsRetriever } from "./control-panel-items-retriever";
+import { Logger } from "../../../common/logger/logger";
+import * as Powershell from "node-powershell";
+
+export class ControlPanelPlugin implements SearchPlugin {
+    public pluginType = PluginType.ControlPanel;
+    private controlPanelItems: ControlPanelItem[] = [];
+
+    constructor(private config: ControlPanelOptions, private readonly logger: Logger) {}
+
+    public isEnabled(): boolean {
+        return this.config.isEnabled;
+    }
+
+    public async execute(searchResultItem: SearchResultItem, privileged: boolean): Promise<void> {
+        const shell = new Powershell({});
+        shell.addCommand(`powershell -Command "Show-ControlPanelItem -Name '${searchResultItem.executionArgument}'"`);
+        shell.invoke()
+            .catch((reason) => this.logger.error("Opening control panel item failed: " + reason))
+            .finally(() => shell.dispose());
+    }
+
+    public async updateConfig(updatedConfig: UserConfigOptions, translationSet: TranslationSet): Promise<void> {
```

Comment:
I would also like to see the pattern that is used in the other parts of ueli.

Instead of this:

``` typescript
public async updateConfig(updatedConfig: UserConfigOptions, translationSet: TranslationSet): Promise<void> {
    this.config = updatedConfig.controlPanelOptions;
}
```

I would rather have this:

``` typescript
public updateConfig(updatedConfig: UserConfigOptions, translationSet: TranslationSet): Promise<void> {
    return new Promise((resolve) => {
        this.config = updatedConfig.controlPanelOptions;
        resolve();
    });
}
```

Also here, I see that your solution is more elegant and uses the more modern `async` keyword. But I would rather have code/pattern consistency throughout all components.